### PR TITLE
add ginkgo label filters for integration tests

### LINKED ISSUE #8004: Use label filters to make it easier to run tests for specific use cases
**What would you like to be cleaned**:

When I am working on a feature or testing specific controllers, the integration tests are difficult to manage as we can only filter by folder. Ginkgo has an optional to add label-filters so we could target specific test cases across the different suites.

Ideally, we can add a label for each integration test that corresponds to what we are testing. That way, one could run the localqueue integration tests without also running the workload tests.

**Why is this needed**:

Easier to work with integration tests.

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
  Add Ginkgo label filters to integration tests

  This change implements label-based filtering for integration tests,
  making it easier to run specific test subsets without being limited
  to folder-based filtering.

  Implemented label taxonomy:
  - Controllers: controller:workload, controller:localqueue,
    controller:clusterqueue, controller:admissioncheck,
    controller:resourceflavor, controller:provisioning,
    controller:jobframework
  - Job Types: job:batch, job:pod, job:jobset, job:pytorch,
    job:tensorflow, job:mpi, job:paddle, job:xgboost, job:jax,
    job:train, job:ray, job:appwrapper
  - Features: feature:tas, feature:multikueue, feature:provisioning
  - Areas: area:core, area:jobs, area:admissionchecks, area:multikueue

  Added labels to all integration test suites:
  - Core controller tests (workload, localqueue, clusterqueue, etc.)
  - All job controller tests (batch, pod, jobset, kubeflow jobs, ray jobs)
  - Admission check provisioning tests
  - MultiKueue integration tests
  - Job framework tests

  Updated Makefile-test.mk with:
  - Documentation explaining the label taxonomy and usage
  - Example commands showing common filter patterns
  - Convenience make targets for frequent use cases
    (test-integration-localqueue, test-integration-pytorch, etc.)

#### Which issue(s) this PR fixes:

Fixes #8004 

#### Special notes for your reviewer:

  Example usage:
```
    # Run only LocalQueue tests
    INTEGRATION_FILTERS="--label-filter=controller:localqueue" make test-integration
    
    # Run all PyTorch job tests
    make test-integration-pytorch
    
    # Run TAS tests across all suites
    INTEGRATION_FILTERS="--label-filter=feature:tas" make test-integration

    # Run job tests excluding Ray
    INTEGRATION_FILTERS="--label-filter=area:jobs && !job:ray" make test-integration
```
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by PBundyra** [comment]
Nice! Can we add labels for FairSharing and AdmissionFairSharing

**Comment by mimowo** [comment]
> Nice! Can we add labels for FairSharing and AdmissionFairSharing

Agree. I'm ok with giving it a try. Maybe we could also follow up later, and this could be the first PR demonstrating how this can be done?

/approve
Leaving lgtm to @PBundyra .

**Comment by PBundyra** [comment]
> Agree. I'm ok with giving it a try. Maybe we could also follow up later, and this could be the first PR demonstrating how this can be done?

Since we are not rushed by a release I'd prefer to cover this in one PR

**Comment by mimowo** [comment]
> Since we are not rushed by a release I'd prefer to cover this in one PR

Why? Incubating the PR for longer will likely result in conflicts on the way. Also, the "done" criteria aren't clear to me, do you think "fairsharing" and "admissionfairsharing" are the last missing features? I think there are likely more of them, and then the question is, can we build exhausitive list?

**Comment by kannon92** [comment]
I can add those to integrations.

A follow up could be doing something similar for e2e.

File: Makefile-test.mk [hunk 1]
```diff
@@ -86,6 +86,62 @@ test-integration-baseline: test-integration ## Run baseline integration tests fo
...
+test-integration-pytorch: test-integration ## Run PyTorchJob integration tests.
+
+.PHONY: test-integration-ray
+test-integration-ray: INTEGRATION_FILTERS= --label-filter="job:ray"
+test-integration-ray: test-integration ## Run Ray job integration tests.
+
+.PHONY: test-integration-tas
+test-integration-tas: INTEGRATION_FILTERS= --label-filter="feature:tas"
+test-integration-tas: test-integration ## Run Topology-Aware Scheduling integration tests.
+
```
Comment (mimowo) [inline Makefile-test.mk]: I don't have a strong view here, but this part looks very verbose and unnecessary to have a target for each possible feature. 

I would rather just document the recognized options here: https://kueue.sigs.k8s.io/docs/contribution_guidelines/testing/

**Comment by PBundyra** [comment]
> Why? Incubating the PR for longer will likely result in conflicts on the way. Also, the "done" criteria aren't clear to me, do you think "fairsharing" and "admissionfairsharing" are the last missing features? I think there are likely more of them, and then the question is, can we build exhausitive list?

I didn't mean to create an exhaustive list of all features. I'm rather leaning those labels in the best-effort manner and those ones seemed valuable to me

File: Makefile-test.mk (hunk 1 above)
Comment (kannon92) [reply Makefile-test.mk]: Good idea. I removed the make target and adding a section to testing.

I left a section in the Make file for reference when reading the make targets.

**Comment by kannon92** [comment]
> Nice! Can we add labels for FairSharing and AdmissionFairSharing

Updated.

File: Makefile-test.mk [hunk 2]
```diff
@@ -86,6 +102,7 @@ test-integration-baseline: test-integration ## Run baseline integration tests fo
 test-integration-extended: INTEGRATION_FILTERS= --label-filter="slow || redundant"
 test-integration-extended: test-integration ## Run extended integration tests for singlecluster suites.
 
+
```
Comment (mimowo) [inline Makefile-test.mk]: nit revert

File: Makefile-test.mk [hunk 3]
```diff
@@ -314,4 +331,4 @@ test-e2e-kueueviz: setup-e2e-env ## Run end-to-end tests for kueueviz without ru
 	CYPRESS_SCREENSHOTS_FOLDER=$(ARTIFACTS)/cypress/screenshots CYPRESS_VIDEOS_FOLDER=$(ARTIFACTS)/cypress/videos \
 	ARTIFACTS=$(ARTIFACTS) KIND_CLUSTER_NAME=$(KIND_CLUSTER_NAME) PROJECT_DIR=$(PROJECT_DIR)/ \
 	KIND_CLUSTER_FILE="kind-cluster.yaml" IMAGE_TAG=$(IMAGE_TAG) \
-	${PROJECT_DIR}/hack/e2e-kueueviz-backend.sh
\ No newline at end of file
+	${PROJECT_DIR}/hack/e2e-kueueviz-backend.sh
```
Comment (mimowo) [inline Makefile-test.mk:333]: nit revert

File: Makefile-test.mk (hunk 2 above)
Comment (kannon92) [reply Makefile-test.mk]: updated.

File: Makefile-test.mk (hunk 3 above)
Comment (kannon92) [reply Makefile-test.mk:333]: updated.

**Comment by kannon92** [comment]
cc @PBundyra 

Ready for your review.

**Comment by kannon92** [comment]
@PBundyra @mimowo anything outstanding here?

**Comment by mimowo** [comment]
Still LGTM

**Comment by PBundyra** [comment]
/lgtm
Thanks @kannon92
